src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/Common/MatchCounter.cs

```diff
@@ -53,7 +53,22 @@ public void AddRange(IEnumerable<T> items)
         }
     }
 
-    public string ToDebugString() => string.Concat(_counts.Select(v => $"{v.Key}: {v.Value}, "));
+    public MatchCounter<T> Intersect(MatchCounter<T> other)
+    {
+        _ = Throw.IfNull(other, nameof(other));
+        var intersection = new MatchCounter<T>();
+        foreach (var kvp in _counts)
```

Comment:
minor: Consider iterating the smaller of the two collections to reduce the overall number of lookups required.

Reply:
Good call.